addons/project/models/project_project.py

```diff
@@ -463,15 +473,18 @@ def create(self, vals_list):
     def write(self, vals):
         # Here we modify the project's stage according to the selected company (selecting the first
         # stage in sequence that is linked to the company).
-        if self.env.user.has_group('project.group_project_stages') and\
-            vals.get('company_id') not in (None, self.company_id.id) and\
-            self.stage_id.company_id:
-            ProjectStage = self.env['project.project.stage']
-            vals['stage_id'] = ProjectStage.search(
-                [('company_id', 'in', (vals['company_id'], False))],
-                order=f"company_id asc, {ProjectStage._order}",
-                limit=1,
-            ).id
+        if self.env.user.has_group('project.group_project_stages') and vals.get('company_id'):
```

Comment:
```suggestion
        company_id = vals.get('company')
        if self.env.user.has_group('project.group_project_stages') and company_id:
```
bc you use it a lot after